imgix/constants.py

```diff
@@ -8,6 +8,22 @@
 SRCSET_INCREMENT_PERCENTAGE = 8
 SRCSET_MAX_SIZE = 8192
 
+# The percentage by which image width-values are incremented by for
+# use in generating the default set of image widths, i.e.
+# `[IMAGE_MIN_WIDTH, .., IMAGE_MAX_WIDTH]`.
+INCREMENT_PERCENTAGE = 0.08
+
+# Representation of an image with a width of zero. This value is used
+# in validation contexts, i.e. "is the width of the passed or requested
+# image greater than or equal to the 'zero width image'."
+IMAGE_ZERO_WIDTH = 0.0
+
+# The minimum width of a default generated image-width.
+IMAGE_MIN_WIDTH = 100.0
+
+# The maximum width of a default generate image-width.
+IMAGE_MAX_WIDTH = 8192.0
```

Comment:
1. These should actually never be of type `float`. We even take an extra step in [imgix-core-js](https://github.com/imgix/imgix-core-js/blob/37918909da11233d2db61b6e8d549988d2040a26/src/imgix-core-js.js#L218-L219) to take the floor of user-passed values.
2. At the risk of sounding a bit pedantic, I think the variable naming here should be a bit more explicit. `100` is the minimum srcset width generated by default, but it's not the smallest image width that imgix can render. Therefore, `IMAGE_MIN_WIDTH` could be a misleading name. Similarly, `8192` is the maximum srcset width generated by default, but it is also the largest image width we can render. My suggestion is to be more explicit by naming them `SRCSET_MIN_WIDTH` and `SRCSET_MAX_WIDTH`, so as to not risk confusing anyone.

Reply:
In reverse order,

2. Correct. It is **our internal default**. It isn't **The** default. This goes for both `100` and `8192`. This is to define ranges and defaults more concretely. That is our default min is `100` and our default max is `8192`. Anything outside these values represents a **custom width**. I did figure the conversation would start right here. I want to push to keep this as is. `IMAGE_MAX_WIDTH` says exactly what it is––it's the maximum image width we can render. 

1. Why should these never be floats? Maybe this is just a difference of opinion. They aren't currently floats because we want to immediately divide and do other calculations with them. Why not define them as such then only convert them to integers once we require them to be, e.g. right before they're stored in the resolution list.

Reply:
In addition, `IMAGE_WIDTH_ZERO` is an implementation dependency. No _actual_ image width should be zero; this also means we need a consistent way of checking for this. Consider it like `nil` or `0`.

As far as the min here being 100 and the max being 8192, consider that to be the default image width number line. Anything outside this range is **custom**. Does this make sense? My intention was to clearly stake out our **range**, this helps to do that. This **does not** mean a user won't be able to define a width smaller than 100. It means that we have defined what that case is called.

Reply:
```
[IMAGE_WIDTH_ZERO, custom, ... 100, ..., possibly custom, ..., IMAGE_WIDTH_MAX]
```

Reply:
I think we ultimately want the same thing here, but we fall in different places on what to do about it.

1. I want to be explicit when a variable/constant pertains to the `srcset` workflow of this library. Barring that, then some indication that `100` is a `DEFAULT_MIN_WIDTH` so as to not potentially confuse someone looking in. The argument for max width is to just best follow the naming structure decided on for min width. I think you'll agree that self documenting code can be very powerful so I'll add this point: good naming semantics can make a big difference for me when exploring an unfamiliar codebase. Think about if the omission of `SRCSET` from the name will be at all confusing for the next team member to join.
2. It's not a difference of opinion really, our rendering engine does not accept floating point dimensional parameters. Going back to my previous point, I don't want to document a width as initialized to a float because it gives the impression that it can be one. But I can see why you would do it in the first place.
3. I don't actually have an issue with the naming for `IMAGE_WIDTH_ZERO`, I think my point about it being a float still applies though.

Reply:
Okay cool, I'll switch all of these values to be integers. Though, we call `int()` on these. Also a lot of this stuff would bode well in docstrings and comments. You could write some good docs here.

Reply:
```suggestion
IMAGE_ZERO_WIDTH = 0

# The minimum width of a default generated image-width.
IMAGE_MIN_WIDTH = 100

# The maximum width of a default generate image-width.
IMAGE_MAX_WIDTH = 8192
```